CommitValidationTest should ignore the release process

Identified in gitter and at https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=13251&view=logs&j=c902ebb4-c9f8-5f09-4e17-ff78fbbc842e&t=9ca98c81-ff64-58f0-9d03-a23ac1c4a111&l=795 ,

`[ERROR]   CommitValidationTest.testCommitMessageHasProperStructure:183 Commit 97ae29bd44209dc6a7f2082516e4175726a9b77e message: "[maven-release-plugin] prepare for next development iteration\n" `

Admins were an ignored user but now they aren't since a recent update and have to follow the proper commit process.

However, the valid use of the release process should not fail the CommitValidationTest. It is the proper commit messaging we use for releases.

The test needs to be updated to not fail on valid release process commits.

--- a/src/test/java/com/puppycrawl/tools/checkstyle/internal/CommitValidationTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/internal/CommitValidationTest.java
@@ -72,12 +72,15 @@ public class CommitValidationTest {
 
     private static final String ISSUE_COMMIT_MESSAGE_REGEX_PATTERN = "^Issue #\\d+: .*$";
     private static final String PR_COMMIT_MESSAGE_REGEX_PATTERN = "^Pull #\\d+: .*$";
+    private static final String RELEASE_COMMIT_MESSAGE_REGEX_PATTERN =
+            "^\\[maven-release-plugin] .*$";
     private static final String OTHER_COMMIT_MESSAGE_REGEX_PATTERN =
             "^(minor|config|infra|doc|spelling|dependency|supplemental): .*$";
 
     private static final String ACCEPTED_COMMIT_MESSAGE_REGEX_PATTERN =
               "(" + ISSUE_COMMIT_MESSAGE_REGEX_PATTERN + ")|"
               + "(" + PR_COMMIT_MESSAGE_REGEX_PATTERN + ")|"
+              + "(" + RELEASE_COMMIT_MESSAGE_REGEX_PATTERN + ")|"
               + "(" + OTHER_COMMIT_MESSAGE_REGEX_PATTERN + ")";
 
     private static final Pattern ACCEPTED_COMMIT_MESSAGE_PATTERN =
@@ -137,6 +140,19 @@ public class CommitValidationTest {
                 + "Test Test Test Test Test Test Test Test Test Test Test Test Test Test Test Test "
                 + "Test Test Test Test Test Test Test  Test Test Test Test Test Test"))
             .isEqualTo(4);
+    }
+
+    @Test
+    public void testReleaseCommitMessage() {
+        assertWithMessage("should accept release commit message for preparing release")
+                .that(validateCommitMessage("[maven-release-plugin] "
+                        + "prepare release checkstyle-10.8.0"))
+                .isEqualTo(0);
+        assertWithMessage("should accept release commit message for preparing for "
+                + "next development iteration")
+                .that(validateCommitMessage("[maven-release-plugin] prepare for next "
+                        + "development iteration"))
+                .isEqualTo(0);
     }
 
     @Test

